pysat/utils/io.py

```diff
@@ -716,9 +738,15 @@ def load_netcdf_xarray(fnames, strict_meta=False, file_format='NETCDF4',
         data = xr.open_mfdataset(fnames, decode_timedelta=decode_timedelta,
                                  combine='by_coords')
 
-    # TODO(#947) Add conversion for timestamps that may have been treated
-    # incorrectly, including origin and unit.  At the moment this is only done
-    # for Pandas data.
+    # If epoch can be converted to a datetime, convert
+    try:
+        data[epoch_name] = xr.DataArray(pds.to_datetime(data[epoch_name],
+                                                        unit=epoch_unit,
+                                                        origin=epoch_origin),
+                                        coords=data[epoch_name].coords)
+    except (KeyError, ValueError):
+        # Value does not exist or cannot be converted to datetime
+        pass
```

Comment:
Reminder to self: add logger statement here with info from error for troubleshooting

Reply:
Dropped try/except here.  no longer needed.